### ISSUE TITLE
why not fix D while updating G?

### ISSUE THREAD

**Issue by XiaoyuShi97** [ai-tor/PyTorch-CycleGAN#31]
Hi, thx for your nice work.

I have a question about training. When you train discriminator, you use detach which means avoid updating generator. But when you update generator, you do not fix discriminator. Is this a common strategy? Because I think in the original GAN paper, we fix discriminator too.
https://github.com/aitorzip/PyTorch-CycleGAN/blob/67da8f9e2b69bd68763451803c7700aaccc92f18/train#L163

**Comment by XiaoyuShi97** [comment]
Sorry. I see that these two nets use different optimizer.
